providers/providers.go

```diff
@@ -34,7 +37,7 @@ func NewProvider(providerConfig options.Provider) (Provider, error) {
 	case options.ADFSProvider:
 		return NewADFSProvider(providerData, providerConfig.ADFSConfig), nil
 	case options.AzureProvider:
-		return NewAzureProvider(providerData, providerConfig.AzureConfig), nil
+		return NewAzureProvider(providerData, *providerConfig.AzureConfig), nil
```

Comment:
This is ripe for a nil pointer dereference error isn't it? We should check it's not nil, or make it so that the constructor accepts the non-nil

Reply:
> Sorry, I sincerely hope that I do not cause more trouble than the change is worth. 

Not at all!

> Trying to configure the provider of type Azure without passing options for Azure causes a nil dereference error even earlier, that is, when trying to default the nil struct.

For azure specifically, we can make it a non pointer as we aren't defaulting anymore

> I therefor think that the check I threw in should be performed earlier, that is, when parsing the config, to ensure that the Azure-specific part is properly set. Do you have an opinion here?

We will want to check for the nil on the OIDC config still. Even if things fail earlier, we need to make sure the package boundaries are observed, so having checks here and earlier in the code is a good thing